adafruit_portalbase/graphics.py

```diff
@@ -76,11 +77,19 @@ def set_background(self, file_or_color, position=None):
 
         if not file_or_color:
             return  # we're done, no background desired
-        if isinstance(file_or_color, str):  # its a filenme:
-            background = displayio.OnDiskBitmap(file_or_color)
+        if isinstance(file_or_color, str):  # it's a filename:
+            if file_or_color.lower().endswith(".bmp"):
+                background = displayio.OnDiskBitmap(file_or_color)
+                palette = background.pixel_shader
+            elif file_or_color.lower().endswith(".jpg") or file_or_color.lower().endswith(".jpeg"):
+                background, palette = adafruit_imageload.load(file_or_color)
+            elif file_or_color.lower().endswith(".png"):
+                background, palette = adafruit_imageload.load(file_or_color)
```

Comment:
These could be combined into one fewer `elif`, since conditionally-run code is the same? Also I factored out the `.lower()`:
```suggestion
        if isinstance(file_or_color, str):  # it's a filename:
            file_lower = file_or_color.lower()
            if file_lower.endswith(".bmp"):
                background = displayio.OnDiskBitmap(file_or_color)
                palette = background.pixel_shader
            elif file_lower.endswith(".jpg") or file_lowerendswith(".jpeg") or file_lower.endswith(".png"):
                background, palette = adafruit_imageload.load(file_or_color)
```